interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -53,6 +54,38 @@ type Prompt struct {
 	listenerReqs []*listener.Request
 }
 
+// jsonPrompt defines the marshalled json structure of a Prompt.
+type jsonPrompt struct {
+	ID          prompting.IDType       `json:"id"`
+	Timestamp   time.Time              `json:"timestamp"`
+	Snap        string                 `json:"snap"`
+	Interface   string                 `json:"interface"`
+	Constraints *jsonPromptConstraints `json:"constraints"`
+}
+
+// jsonPromptConstraints defines the marshalled json structure of promptConstraints.
+type jsonPromptConstraints struct {
+	Path                 string   `json:"path"`
+	RequestedPermissions []string `json:"requested-permissions"`
+	AvailablePermissions []string `json:"available-permissions"`
+}
+
+func (p *Prompt) MarshalJSON() ([]byte, error) {
```

Comment:
I still think we should have a MarshalForClient method instead because of the listenerReqs field, I will put here a TODO.